src/main/java/chobo/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package chobo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+/**
+ * Represents a task that needs to be done before a specific date/time.
+ * A chobo.Deadline object corresponds to a task with a deadline
+ * in the Chobo chatbot.
+ */
+public class Deadline extends Task {
+    private String unformattedDate;
+    private LocalDateTime by;
+
+    /**
+     * Creates a new chobo.Deadline task.
+     *
+     * @param name The description of the task.
+     * @param done The status of the task (true if done, false otherwise).
+     * @param date The deadline of the task.
+     */
+    public Deadline(String name, boolean done, String date) {
+        super(name, done);
+        this.unformattedDate = date;
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d-M-yyyy HHmm");
+        this.by = LocalDateTime.parse(date.trim(), dateTimeFormatter);
```

Comment:
Consider using name deadline instead of by for deadline date so that it might be more understandable